### DISCUSSION TITLE
install Elastalert2

### DISCUSSION THREAD

**Discussion by Firasbg50** [jertel/elastalert2#950] [Q&A] [answered]
it's needed ubuntu just 21. x  to work ?  or 22. x ubuntu can work also because it has python3.10.5

**Comment by jertel** [comment, marked as answer]
It should work on Ubuntu 22.x. If there's a problem, let us know so we can investigate.

**Reply by Firasbg50** [reply]
ok @jertel i will

**Reply by Firasbg50** [reply]
@jertel see this warning when I start the installation like the guide using this cmd : "pip3 install elastalert2" l
  does it have an effect on the installation??
 ```
 WARNING: The script pyjwt is installed in '/home/elastic/.local/bin' which is not on PATH.
  Consider adding this directory to PATH or, if you prefer to suppress this warning, use --no-warn-script-location.
  WARNING: The script stomp is installed in '/home/elastic/.local/bin' which is not on PATH.
  Consider adding this directory to PATH or, if you prefer to suppress this warning, use --no-warn-script-location.
  WARNING: The script jsonschema is installed in '/home/elastic/.local/bin' which is not on PATH.
  Consider adding this directory to PATH or, if you prefer to suppress this warning, use --no-warn-script-location.
  WARNING: The script normalizer is installed in '/home/elastic/.local/bin' which is not on PATH.
  Consider adding this directory to PATH or, if you prefer to suppress this warning, use --no-warn-script-location.
  WARNING: The script jirashell is installed in '/home/elastic/.local/bin' which is not on PATH.
  Consider adding this directory to PATH or, if you prefer to suppress this warning, use --no-warn-script-location.
  WARNING: The scripts elastalert, elastalert-create-index and elastalert-test-rule are installed in '/home/elastic/.local/bin' which is not on PATH.
  Consider adding this directory to PATH or, if you prefer to suppress this warning, use --no-warn-script-location.
Successfully installed Jinja2-3.1.2 PyJWT-1.7.1 PyYAML-6.0 apscheduler-3.9.1 aws-requests-auth-0.4.3 boto3-1.24.75 botocore-1.27.75 cffi-1.15.1 charset-normalizer-2.1.1 croniter-1.3.7 defusedxml-0.7.1 docopt-0.6.2 elastalert2-2.6.0 elasticsearch-7.10.1 envparse-0.2.0 exotel-0.1.5 jira-3.4.1 jmespath-1.0.1 jsonpointer-2.3 jsonschema-4.16.0 packaging-21.3 prison-0.2.1 prometheus-client-0.14.1 py-zabbix-1.1.7 pycparser-2.21 python-dateutil-2.8.2 requests-2.28.1 requests-oauthlib-1.3.1 requests-toolbelt-0.9.1 s3transfer-0.6.0 sortedcontainers-2.4.0 statsd-tags-3.2.1.post1 stomp.py-8.0.1 tencentcloud-sdk-python-3.0.733 texttable-1.6.4 twilio-6.57.0 typing-extensions-4.3.0 tzlocal-2.1


```

**Comment by Firasbg50** [comment]
this warning also : 

root@elastic:/home/elastic# pip install elastalert2
Requirement already satisfied: elastalert2 in /usr/local/lib/python3.10/dist-packages (2.6.0)
Requirement already satisfied: jsonpointer>=2.2 in /usr/local/lib/python3.10/dist-packages (from elastalert2) (2.3)
Requirement already satisfied: stomp.py>=8.0.1 in /usr/local/lib/python3.10/dist-packages (from elastalert2) (8.0.1)
Requirement already satisfied: py-zabbix>=1.1.7 in /usr/local/lib/python3.10/dist-packages (from elastalert2) (1.1.7)
Requirement already satisfied: prometheus-client>=0.13.1 in /usr/local/lib/python3.10/dist-packages (from elastalert2) (0.14.1)
Requirement already satisfied: cffi>=1.15.0 in /usr/local/lib/python3.10/dist-packages (from elastalert2) (1.15.1)
Requirement already satisfied: requests>=2.27.1 in /usr/local/lib/python3.10/dist-packages (from elastalert2) (2.28.1)
Requirement already satisfied: exotel>=0.1.5 in /usr/local/lib/python3.10/dist-packages (from elastalert2) (0.1.5)
Requirement already satisfied: sortedcontainers>=2.4.0 in /usr/local/lib/python3.10/dist-packages (from elastalert2) (2.4.0)
Requirement already satisfied: texttable>=1.6.4 in /usr/local/lib/python3.10/dist-packages (from elastalert2) (1.6.4)
Requirement already satisfied: croniter>=1.2.0 in /usr/local/lib/python3.10/dist-packages (from elastalert2) (1.3.7)
Requirement already satisfied: tencentcloud-sdk-python>=3.0.577 in /usr/local/lib/python3.10/dist-packages (from elastalert2) (3.0.733)
Requirement already satisfied: jsonschema>=4.4.0 in /usr/local/lib/python3.10/dist-packages (from elastalert2) (4.16.0)
Requirement already satisfied: envparse>=0.2.0 in /usr/local/lib/python3.10/dist-packages (from elastalert2) (0.2.0)
Requirement already satisfied: PyYAML>=6.0 in /usr/local/lib/python3.10/dist-packages (from elastalert2) (6.0)
Requirement already satisfied: elasticsearch==7.10.1 in /usr/local/lib/python3.10/dist-packages (from elastalert2) (7.10.1)
Requirement already satisfied: python-dateutil>=2.8.2 in /usr/local/lib/python3.10/dist-packages (from elastalert2) (2.8.2)
Requirement already satisfied: prison>=0.2.1 in /usr/local/lib/python3.10/dist-packages (from elastalert2) (0.2.1)
Requirement already satisfied: tzlocal==2.1 in /usr/local/lib/python3.10/dist-packages (from elastalert2) (2.1)
Requirement already satisfied: aws-requests-auth>=0.4.3 in /usr/local/lib/python3.10/dist-packages (from elastalert2) (0.4.3)
Requirement already satisfied: twilio==6.57.0 in /usr/local/lib/python3.10/dist-packages (from elastalert2) (6.57.0)
Requirement already satisfied: boto3>=1.20.53 in /usr/local/lib/python3.10/dist-packages (from elastalert2) (1.24.75)
Requirement already satisfied: apscheduler<4.0,>=3.8.1 in /usr/local/lib/python3.10/dist-packages (from elastalert2) (3.9.1)
Requirement already satisfied: Jinja2==3.1.2 in /usr/local/lib/python3.10/dist-packages (from elastalert2) (3.1.2)
Requirement already satisfied: statsd-tags==3.2.1.post1 in /usr/local/lib/python3.10/dist-packages (from elastalert2) (3.2.1.post1)
Requirement already satisfied: jira>=3.1.1 in /usr/local/lib/python3.10/dist-packages (from elastalert2) (3.4.1)
Requirement already satisfied: urllib3<2,>=1.21.1 in /usr/lib/python3/dist-packages (from elasticsearch==7.10.1->elastalert2) (1.26.5)
Requirement already satisfied: certifi in /usr/lib/python3/dist-packages (from elasticsearch==7.10.1->elastalert2) (2020.6.20)
Requirement already satisfied: MarkupSafe>=2.0 in /usr/lib/python3/dist-packages (from Jinja2==3.1.2->elastalert2) (2.0.1)
Requirement already satisfied: PyJWT==1.7.1 in /usr/local/lib/python3.10/dist-packages (from twilio==6.57.0->elastalert2) (1.7.1)
Requirement already satisfied: pytz in /usr/lib/python3/dist-packages (from twilio==6.57.0->elastalert2) (2022.1)
Requirement already satisfied: six in /usr/lib/python3/dist-packages (from twilio==6.57.0->elastalert2) (1.16.0)
Requirement already satisfied: setuptools>=0.7 in /usr/lib/python3/dist-packages (from apscheduler<4.0,>=3.8.1->elastalert2) (59.6.0)
Requirement already satisfied: jmespath<2.0.0,>=0.7.1 in /usr/local/lib/python3.10/dist-packages (from boto3>=1.20.53->elastalert2) (1.0.1)
Requirement already satisfied: botocore<1.28.0,>=1.27.75 in /usr/local/lib/python3.10/dist-packages (from boto3>=1.20.53->elastalert2) (1.27.75)
Requirement already satisfied: s3transfer<0.7.0,>=0.6.0 in /usr/local/lib/python3.10/dist-packages (from boto3>=1.20.53->elastalert2) (0.6.0)
Requirement already satisfied: pycparser in /usr/local/lib/python3.10/dist-packages (from cffi>=1.15.0->elastalert2) (2.21)
Requirement already satisfied: requests-toolbelt in /usr/local/lib/python3.10/dist-packages (from jira>=3.1.1->elastalert2) (0.9.1)
Requirement already satisfied: typing-extensions>=3.7.4.2 in /usr/local/lib/python3.10/dist-packages (from jira>=3.1.1->elastalert2) (4.3.0)
Requirement already satisfied: defusedxml in /usr/local/lib/python3.10/dist-packages (from jira>=3.1.1->elastalert2) (0.7.1)
Requirement already satisfied: requests-oauthlib>=1.1.0 in /usr/local/lib/python3.10/dist-packages (from jira>=3.1.1->elastalert2) (1.3.1)
Requirement already satisfied: packaging in /usr/local/lib/python3.10/dist-packages (from jira>=3.1.1->elastalert2) (21.3)
Requirement already satisfied: attrs>=17.4.0 in /usr/lib/python3/dist-packages (from jsonschema>=4.4.0->elastalert2) (21.2.0)
Requirement already satisfied: pyrsistent!=0.17.0,!=0.17.1,!=0.17.2,>=0.14.0 in /usr/lib/python3/dist-packages (from jsonschema>=4.4.0->elastalert2) (0.18.1)
Requirement already satisfied: charset-normalizer<3,>=2 in /usr/local/lib/python3.10/dist-packages (from requests>=2.27.1->elastalert2) (2.1.1)
Requirement already satisfied: idna<4,>=2.5 in /usr/lib/python3/dist-packages (from requests>=2.27.1->elastalert2) (3.3)
Requirement already satisfied: docopt<0.7.0,>=0.6.2 in /usr/local/lib/python3.10/dist-packages (from stomp.py>=8.0.1->elastalert2) (0.6.2)
Requirement already satisfied: oauthlib>=3.0.0 in /usr/lib/python3/dist-packages (from requests-oauthlib>=1.1.0->jira>=3.1.1->elastalert2) (3.2.0)
Requirement already satisfied: pyparsing!=3.0.5,>=2.0.2 in /usr/lib/python3/dist-packages (from packaging->jira>=3.1.1->elastalert2) (2.4.7)
`WARNING: Running pip as the 'root' user can result in broken permissions and conflicting behaviour with the system package manager. It is recommended to use a virtual environment instead: https://pip.pypa.io/warnings/venv`

**Reply by jertel** [reply]
Those warnings should not interfere with ElastAlert 2. The last warning about running as root will not cause a problem but it is not best practice to run programs as root. I suggest using the ElastAlert2 Docker container if possible. This avoids all the warnings and the root issue.

**Reply by Firasbg50** [reply]
I use ubuntu because I do my end-of-study project with ubuntu

**Reply by jertel** [reply]
If this is for an academic study then you can likely ignore the "running pip as root" warning.

**Reply by Firasbg50** [reply]
Error when sending email alert 
elastic@elastic:~/elastalert2$ python3 -m elastalert.elastalert --verbose --rule windows_login_fail.yaml
INFO:elastalert:1 rules loaded
INFO:elastalert:Starting up
INFO:elastalert:Disabled rules are: []
INFO:elastalert:Sleeping for 59.9998 seconds
INFO:elastalert:Queried rule lOGIN FAILED WINDOWS from 2022-09-17 21:39 UTC to 2022-09-17 21:42 UTC: 3 / 3 hits
`ERROR:elastalert:Error while running alert email: Error connecting to SMTP host: (534, b'5.7.9 Application-specific password required. Learn more at\n5.7.9 ` https://support.google.com/mail/?p=InvalidSecondFactor e4-20020a05600c4e4400b003a6a3595edasm7288673wmq.27 - gsmtp')
INFO:elastalert:Ignoring match for silenced rule lOGIN FAILED WINDOWS
INFO:elastalert:Ignoring match for silenced rule lOGIN FAILED WINDOWS
INFO:elastalert:Ran lOGIN FAILED WINDOWS from 2022-09-17 21:39 UTC to 2022-09-17 21:42 UTC: 3 query hits (0 already seen), 3 matches, 0 alerts sent
INFO:elastalert:lOGIN FAILED WINDOWS range 173

**Reply by Firasbg50** [reply]
in the rule that i did "windows_login_fail.yaml" i configure 
```
# (Required)
# A list of Elasticsearch filters used for find events
# These filters are joined with AND and nested in a filtered query
# For more info: https://www.elastic.co/guide/en/elasticsearch/reference/current/>
filter:
- query_string:
    query: "event.code:7036"

# (Required)
# The alert is use when a match is found
alert:
- "email"

# (required, email specific)
# a list of email addresses to send alerts to
email:
- "bougrine.firas@gmail.com"
smtp_host: "smtp.gmail.com"
smtp_port: 465
smtp_ssl: true
from_addr: "bougrine.firas@gmail.com"
smtp_auth_file: "smtp_auth_file.yaml"
```

**Reply by jertel** [reply]
See https://github.com/jertel/elastalert2/discussions/524.

**Reply by Firasbg50** [reply]
now ok this error not appear but other error appeared 
ERROR:elastalert:Error while running alert email: Error connecting to SMTP host: [Errno 101] Network is unreachable

**Reply by jertel** [reply]
Sounds like your Ubuntu machine lost Internet connectivity. You will need to figure this one out on your own, as it is not ElastAlert 2 related.

**Reply by Firasbg50** [reply]
the solution is there , is the create an app password and to let the port 465 , it's works 
https://github.com/Yelp/elastalert/issues/627
